core/src/upgrade.rs

```diff
@@ -63,19 +63,21 @@ mod either;
 mod error;
 mod pending;
 mod ready;
+mod secure;
 mod select;
 
+pub use self::{
+    denied::DeniedUpgrade, pending::PendingUpgrade, ready::ReadyUpgrade, select::SelectUpgrade,
+};
+pub use crate::Negotiated;
 pub(crate) use apply::{
     apply, apply_inbound, apply_outbound, InboundUpgradeApply, OutboundUpgradeApply,
 };
 pub(crate) use error::UpgradeError;
 use futures::future::Future;
-
-pub use self::{
-    denied::DeniedUpgrade, pending::PendingUpgrade, ready::ReadyUpgrade, select::SelectUpgrade,
-};
-pub use crate::Negotiated;
+use libp2p_identity::PeerId;
 pub use multistream_select::{NegotiatedComplete, NegotiationError, ProtocolError, Version};
+pub(crate) use secure::{secure, EitherSecurityFuture};
```

Comment:
Can we please keep the grouping of `pub use`, `pub(crate)` and `use`.

Reply:
Of course, no problem.